### PR TITLE
test() without testthat on search path

### LINKED ISSUE #798: Figure out how to eliminate require(testthat)
as requested by CRAN

### PR DESCRIPTION
**Pull request by krlmlr**
Fixes #798.

### CODE REVIEW

**Comment by krlmlr** [comment]
Should work now. The implementation hacks into `pkg$depends` and adds `testthat` as last dependency. (Not sure about that -- perhaps it should be the *first* dependency.)

**Comment by krlmlr** [comment]
Sorry, not yet. Travis doesn't show errors with R CMD check: https://travis-ci.org/hadley/devtools/builds/64069778#L1615.

**Comment by krlmlr** [comment]
Ready for review. Includes two other pull requests, which can also be merged separately. Note that Travis build status seems unreliable (https://github.com/travis-ci/travis-ci/issues/3697).

**Comment by hadley** [comment]
Would you mind also updating cran-comments?

**Comment by krlmlr** [comment]
What should go into cran-comments?

**Comment by hadley** [comment]
You can remove the first bullet point

**Comment by krlmlr** [comment]
Updated cran-comments, tests still running.

File: R/test.r [hunk 1]
```diff
@@ -37,7 +37,10 @@ test <- function(pkg = ".", filter = NULL, ...) {
   }
 
   # Need to attach testthat so that (e.g.) context() is available
-  library(testthat, quietly = TRUE)
+  # Update package dependency to avoid explicit require() call (#798)
+  pkg$depends <- paste0("testthat, ", pkg$depends)
```
Comment (hadley) [inline R/test.r]: I wonder if the absence of `testthat` should just be an error.

**Comment by hadley** [comment]
Could you please merge/rebase to fix the travis error?

File: R/test.r (hunk 1 above)
Comment (krlmlr) [inline R/test.r]: Could you please elaborate? `testthat` is normally listed in `Suggests:` and by default would not be loaded by `load_all`.

**Comment by krlmlr** [comment]
Done, tests pass now.

File: R/test.r (hunk 1 above)
Comment (hadley) [inline R/test.r]: Oh, oops, I was getting confused.

**Comment by hadley** [comment]
Thanks!

**Comment by hadley** [comment]
Could you please take a look at http://win-builder.r-project.org/r8sssx3TY6ue/00check.log ? This seems to be failing on win builder on r-devel

**Pull request merged**